File: 프로그래머스/2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.py
def solution(str1, str2):
    strset1 = []
    for i in range(0,len(str1)-1):
        if str1[i:i+2].isalpha():
            strset1.append(str(str1[i:i+2]).upper())
            
    strset2 = []
    for i in range(0,len(str2)-1):
        if str2[i:i+2].isalpha():
            strset2.append(str(str2[i:i+2]).upper())
            
    s1 = strset1.copy()
    s2 = strset2.copy()
            
    interset = []
    
    for i in strset1:
        if i in s2 :
            interset.append(i)
            s2.remove(i)
            s1.remove(i)
    
    unionset = []
    
    unionset = s1+s2+interset

    print(unionset,interset)
    answer = 0 
    if len(unionset) == 0 and len(interset) == 0:
        answer = 1*65536
    else :
        answer = int(len(interset)/len(unionset)*65536)
        
        
    return answer
                
  
                

File: 프로그래머스/2/17677. ［1차］ 뉴스 클러스터링/［1차］ 뉴스 클러스터링.java
import java.util.*;

class Solution {
    public List<Character> english = new ArrayList<>(Arrays.asList('a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'));
    public int solution(String str1, String str2) {
        int answer = 0;
        List<String> str1list = new ArrayList<>();
for (int i = 1; i < str1.length(); i++) {
    String pair = (str1.charAt(i-1) + "" + str1.charAt(i)).toLowerCase();
    if (pair.matches("[a-z]{2}")) { 
        str1list.add(pair);
    }
}

List<String> str2list = new ArrayList<>();
for (int i = 1; i < str2.length(); i++) {
    String pair = (str2.charAt(i-1) + "" + str2.charAt(i)).toLowerCase();
    if (pair.matches("[a-z]{2}")) {
        str2list.add(pair);
    }
}
        
        
        if (str1list.size() == 0 && str2list.size() == 0)
        {
            return 65536;
        } 
        
        if (str1list.size() == 0 || str2list.size() == 0)
        {
            return 0;
        }
        
        List<String> samelist = new ArrayList<>();
        List<String> list2save = new ArrayList<>(str2list);
        
        
        for (int i = 0 ;i<str1list.size(); i++)
        {
            if (list2save.contains(str1list.get(i)))
            {
                samelist.add(str1list.get(i));
                list2save.remove(str1list.get(i));
            }
        }
        
        if (samelist.size() ==0 )
        {
            return 0;
        }
        
        
        double val = (double) samelist.size() / (str1list.size()+str2list.size()-samelist.size());
        
        //기호,숫자 제거 
        return (int)(val*65536);
    }
}